File: library/src/main/java/nl/endevelopment/r2randroid/r2rlib/models/DayFlags.java
package nl.endevelopment.r2randroid.r2rlib.models;

/**
 * Created by jan on 11/07/16.
 */

public class DayFlags {
    public static final int SUNDAY = 1;
    public static final int MONDAY = 2;
    public static final int TUESDAY = 4;
    public static final int WEDNESDAY = 8;
    public static final int THURSDAY = 16;
    public static final int FRIDAY = 32;
    public static final int SATURDAY = 64;

    public static final int EVERYDAY = SUNDAY + MONDAY + TUESDAY + WEDNESDAY + THURSDAY + FRIDAY + SATURDAY;
    public static final int WEEKENDS = SATURDAY + SUNDAY;
    public static final int WEEKDAYS = MONDAY + TUESDAY + WEDNESDAY + THURSDAY + FRIDAY;
    public static final int NEVER = 0;


    private int flag;

    public DayFlags(int flag) {
        this.flag = flag;
    }

    public int getFlag() {
        return flag;
    }

    public void setFlag(int flag) {
        this.flag = flag;
    }


    /**
     * is given day flagged
     * use static integer from this class
     * @param day static int above
     * @return boolean
     */
    public boolean isDay(int day) {
        return (flag & day) == day;
    }


    @Override
    public String toString() {
        return "DayFlags{" +
                "flag=" + flag +
                '}';
    }
}

File: library/src/main/java/com/getroadmap/r2rlib/models/DayFlags.kt
package com.getroadmap.r2rlib.models

/**
 * Created by jan on 11/07/16.
 */

class DayFlags(var flag: Int) {


    /**
     * is given day flagged
     * use static integer from this class
     * @param day static int above
     * @return boolean
     */
    fun isDay(day: Int): Boolean {
        return flag and day == day
    }


    override fun toString(): String {
        return "DayFlags{" +
                "flag=" + flag +
                '}'.toString()
    }

    companion object {
        @JvmField val SUNDAY = 1
        @JvmField val MONDAY = 2
        @JvmField val TUESDAY = 4
        @JvmField val WEDNESDAY = 8
        @JvmField val THURSDAY = 16
        @JvmField val FRIDAY = 32
        @JvmField val SATURDAY = 64

        @JvmField val EVERYDAY = SUNDAY + MONDAY + TUESDAY + WEDNESDAY + THURSDAY + FRIDAY + SATURDAY
        @JvmField val WEEKENDS = SATURDAY + SUNDAY
        @JvmField val WEEKDAYS = MONDAY + TUESDAY + WEDNESDAY + THURSDAY + FRIDAY
        @JvmField val NEVER = 0
    }
}
